src/main/kotlin/org/jetbrains/intellij/dependency/CustomPluginsRepository.kt

```diff
@@ -66,22 +63,22 @@ class CustomPluginsRepository(repositoryUrl: String) : PluginsRepository {
         return downloadZipArtifact(project, downloadUrl, plugin)
     }
 
-    private fun getCacheDirectoryPath(project: Project): String {
-        // todo: a better way to define cache directory
-        val gradleHomePath = project.gradle.gradleUserHomeDir.absolutePath
-        val mavenCacheDirectoryPath = Paths.get(gradleHomePath, "caches/modules-2/files-2.1").toString()
-        val digest = MessageDigest.getInstance("SHA-1").digest(repositoryUrl.toByteArray())
-        val hash = Hex.encodeHex(digest).toString()
-        return Paths.get(mavenCacheDirectoryPath, "com.jetbrains.intellij.idea", hash).toString()
-    }
+    private fun downloadZipArtifact(project: Project, url: String, plugin: PluginDependencyNotation): File? {
+        val repository = project.repositories.ivy { ivy ->
+            ivy.url = URI(url)
```

Comment:
Just wondering: is the repository declared only once or does it get declared on every execution?